src/compiler/checker.ts

```diff
@@ -8977,17 +8978,35 @@ namespace ts {
                 }
             }
 
-            function typeRelatedToEachType(source: Type, target: UnionOrIntersectionType, reportErrors: boolean): Ternary {
+            function typeRelatedToEachType(source: Type, target: IntersectionType, reportErrors: boolean): Ternary {
                 let result = Ternary.True;
                 const targetTypes = target.types;
+                const saveDisableWeakTypeErrors = disableWeakTypeErrors;
+                disableWeakTypeErrors = true;
                 for (const targetType of targetTypes) {
                     const related = isRelatedTo(source, targetType, reportErrors);
                     if (!related) {
+                        disableWeakTypeErrors = saveDisableWeakTypeErrors;
                         return Ternary.False;
                     }
                     result &= related;
                 }
-                return result;
+                disableWeakTypeErrors = saveDisableWeakTypeErrors;
+                return reportAssignmentToWeakIntersection(source, target, reportErrors) ? Ternary.False : result;

```

Comment:
I assume that you always want to check the combined intersection type. Put another way, I assume that the only reason that `disableWeakTypeErrors` would be false is to prevent checking of individual intersection constituents, which doesn't apply to `reportAssignmentToWeakIntersection`. 

tl;dr It doesn't need to check `disableWeakTypeErrors` because it doesn't apply.